relay-filter/src/legacy_browsers.rs

```diff
@@ -2,82 +2,71 @@
 
 use std::collections::BTreeSet;
 
-use relay_event_schema::protocol::Event;
 use relay_ua::UserAgent;
 
-use crate::{FilterStatKey, LegacyBrowser, LegacyBrowsersFilterConfig};
+use crate::{FilterStatKey, Filterable, LegacyBrowser, LegacyBrowsersFilterConfig};
 
 /// Checks if the event originates from legacy browsers.
-pub fn matches(event: &Event, browsers: &BTreeSet<LegacyBrowser>) -> bool {
-    if let Some(user_agent_string) = event.user_agent() {
-        let user_agent = relay_ua::parse_user_agent(user_agent_string);
+fn matches(user_agent: &str, browsers: &BTreeSet<LegacyBrowser>) -> bool {
```

Comment:
I made the `matches` functions private everywhere, they do not need to be public.